packages/stark-ui/src/modules/keyboard-directives/directives/transform-input.directive.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { Component, OnInit } from "@angular/core";
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { FormControl, FormsModule, ReactiveFormsModule } from "@angular/forms";
+import { By } from "@angular/platform-browser";
+import { MockStarkLoggingService } from "@nationalbankbelgium/stark-core/testing";
+import { STARK_LOGGING_SERVICE } from "@nationalbankbelgium/stark-core";
+import { StarkTransformInputDirective } from "./transform-input.directive";
+
+/**
+ * @ignore
+ * Mocks an InputEvent on the element with the given value.
+ * @param value - the value to set the element to.
+ * @param element - the HTMLInputElement to mock the event on
+ */
+const mockInputEvent: (value: string, element: HTMLInputElement) => void = (value: string, element: HTMLInputElement): void => {
+	const event: Event = document.createEvent("Event");
+	event.initEvent("input", true, true);
+	element.value = value;
+	element.dispatchEvent(event);
+};
+
+describe("TransformsInputDirective with ngModel", () => {
+	/**
+	 * @ignore
+	 * mocked wrapper for the directive
+	 */
+	@Component({
+		selector: "test-component",
+		template: "<input [(ngModel)]='value' starkTransformInput='upper'/>"
+	})
+	class TestComponent {
```

Comment:
You can move this TestComponent to the root `describe` and create a function to get the template to render so you can re-use it in the different `describe`s...

We do this for example in the `on-enter-key.directive.spec.ts` ;)